docs/REST/TEC/V1/interfaces.md

```diff
@@ -24,54 +24,59 @@ Base interface that all endpoints must implement. Defines:
 
 For endpoints that support GET requests:
 
-- `read(WP_REST_Request $request): WP_REST_Response`
-- `read_args(): Collection`
-- `read_schema(): OpenAPI_Schema`
-- `can_read(WP_REST_Request $request): bool`
+- `read(array $params = []): WP_REST_Response` - Handle read operation with sanitized parameters
+- `can_read(WP_REST_Request $request): bool` - Permission check for read operations
+- `read_args(): QueryArgumentCollection` - Define query parameters for GET requests
+- `read_schema(): OpenAPI_Schema` - OpenAPI documentation for read operation
+- `get_read_attributes(): array` - WordPress REST registration attributes
 
 #### 3. `Creatable_Endpoint`
 
 **Location**: `Creatable_Endpoint.php`
 
 For endpoints that support POST requests:
 
-- `create(WP_REST_Request $request): WP_REST_Response`
-- `create_args(): Collection`
-- `create_schema(): OpenAPI_Schema`
-- `can_create(WP_REST_Request $request): bool`
+- `create(array $params = []): WP_REST_Response` - Handle create operation with sanitized parameters
+- `can_create(WP_REST_Request $request): bool` - Permission check for create operations
+- `create_args(): RequestBodyCollection` - Define request body structure (changed from QueryArgumentCollection)
+- `create_schema(): OpenAPI_Schema` - OpenAPI documentation for create operation
+- `get_create_attributes(): array` - WordPress REST registration attributes
 
 #### 4. `Updatable_Endpoint`
 
 **Location**: `Updatable_Endpoint.php`
 
 For endpoints that support PUT/PATCH requests:
 
-- `update(WP_REST_Request $request): WP_REST_Response`
-- `update_args(): Collection`
-- `update_schema(): OpenAPI_Schema`
-- `can_update(WP_REST_Request $request): bool`
+- `update(array $params = []): WP_REST_Response` - Handle update operation with sanitized parameters
+- `can_update(WP_REST_Request $request): bool` - Permission check for update operations
+- `update_args(): RequestBodyCollection` - Define request body structure (changed from QueryArgumentCollection)
```

Comment:
Here and elsewhere it look like this method should use `params` over `args`.
Parameters are those in the signature (what the method seems to be returning) while arguments are those in the concrete call.